### DISCUSSION TITLE
v2.12.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#2796] [General]
### User Journey Report

First up, we have the long-awaited User Journey report which lets you visually see how users navigate through your website. You can configure whether you want to see all user paths or start/end with a certain page or action.

![image](https://github.com/umami-software/umami/assets/477975/59ed2f34-481d-403d-9139-0b85ff650f5e)

### Goals Report

Then we have the new Goals report. With this report you can set goals for pageviews, events and even event data.

![image](https://github.com/umami-software/umami/assets/477975/b35d13f0-1f0b-4385-a7b6-6427823f9e1d)

### Insights report

The Insights report now displays columns for **Visits**, **Bounce rate**, and **Visit duration**.

![image](https://github.com/umami-software/umami/assets/477975/baf5eaec-9a20-438f-8189-229d7070aa57)

### Collapse menu

On all reports, there is a new tab in the reports menu that lets you collapse the menu so you have more room to view your reports.

![image](https://github.com/umami-software/umami/assets/477975/6aca12f5-4a7e-4911-838a-88ebb63115fa)

### Comparison Mode

We added a new mode to the details screen which let's you compare metrics across two date periods. You will see the data overlayed over each other so you can observe trends occurring in your traffic.

![image](https://github.com/umami-software/umami/assets/477975/bc83e952-00e6-4005-8a19-a82b67b8d29d)

Additionally you can view the changes across all your different metrics.

![image](https://github.com/umami-software/umami/assets/477975/0276e3c0-9b85-4287-bd63-0a4a27eff334)

### Updated Metrics Header

We've made a few cosmetic changes to the website header. Now all the change stats are listed as percentages with an arrow to clearly indicate the change direction. You can still hover over the percent value to get the actual change value.

![image](https://github.com/umami-software/umami/assets/477975/c3673936-abaf-4c2b-8574-56267c4aff5e)

### Hosts Filtering

For users who track stats across multiple domains or subdomains, you can now filter by hostname.

<img src="https://github.com/umami-software/umami/assets/477975/657c71d0-2416-444f-b5fd-3e62ff880afa" width="600" />

## Fixes
- Fixed issue with metrics not filtering #2737 
- Fixed issue with `offset` query parameter #2752 
- Fixed issue payload size in realtime page #2747
- Refactored realtime page to be more accurate #2754
- Ensure websiteId is always sent from tracker #2614

## Updates
- Language updates: Ukrainian, Bulgarian, Brazilian Portuguese, Polish

## Thank you
@giuseongit @leits @frontedu @shubhusion @abahchevanov @frantic 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.12.0'>v2.12.0</a>.</em>

**Comment by giuliomagnifico** [comment]
Hi, I'm getting this error after the update, I'm no expert with docker, how can i fix it?

Thanks

```
root@analytics:/opt/umami# docker-compose up -d
Recreating 6c62bf8f5d8e_umami-db-1 ...

ERROR: for 6c62bf8f5d8e_umami-db-1  'ContainerConfig'

ERROR: for db  'ContainerConfig'
Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 203, in perform_command
  File "compose/metrics/decorator.py", line 18, in wrapper
  File "compose/cli/main.py", line 1186, in up
  File "compose/cli/main.py", line 1182, in up
  File "compose/project.py", line 702, in up
  File "compose/parallel.py", line 108, in parallel_execute
  File "compose/parallel.py", line 206, in producer
  File "compose/project.py", line 688, in do
  File "compose/service.py", line 581, in execute_convergence_plan
  File "compose/service.py", line 503, in _execute_convergence_recreate
  File "compose/parallel.py", line 108, in parallel_execute
  File "compose/parallel.py", line 206, in producer
  File "compose/service.py", line 496, in recreate
  File "compose/service.py", line 615, in recreate_container
  File "compose/service.py", line 334, in create_container
  File "compose/service.py", line 922, in _get_container_create_options
  File "compose/service.py", line 962, in _build_container_volume_options
  File "compose/service.py", line 1549, in merge_volume_bindings
  File "compose/service.py", line 1579, in get_container_data_volumes
KeyError: 'ContainerConfig'
[532413] Failed to execute script docker-compose
```

**Edit: fixed it** by simply rebuild/restart the container:

```
root@analytics:/opt/umami# docker-compose down
Stopping umami-umami-1 ... done
Removing umami-umami-1           ... done
Removing 6c62bf8f5d8e_umami-db-1 ... done
Removing network umami_default
root@analytics:/opt/umami# docker-compose up --build -d
Creating network "umami_default" with the default driver
Creating umami_db_1 ... done
Creating umami_umami_1 ... done
```

**Reply by Joedmin** [reply]
I use only the image from the container registry and restarting it does not fix it 😔

**Reply by Joedmin** [reply]
I found out it was uBlock Origin issue, disabling it fot my site worked!

**Comment by mikeckennedy** [comment]
Fantastic update, thank you all!

**Comment by jovylle** [comment]
Exciting update!

is there a way to filter our display table or graphs based on a custom data

**Comment by bgervan** [comment]
Looks like a cool update. Will you include these features in the demo?
